### PR TITLE
[mui-datatables] add typings for style overrides

### PR DESCRIPTION
**Pull request by petrbela**
Resolves discussion in https://github.com/gregnb/mui-datatables/issues/580

Checklist
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [MUIDataTable.js](https://github.com/gregnb/mui-datatables/blob/master/src/MUIDataTable.js) and individual [components](https://github.com/gregnb/mui-datatables/tree/master/src/components)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

---

- [x] Add `@material-ui/core` to [dependenciesWhitelist.txt](https://github.com/microsoft/types-publisher/blob/master/dependenciesWhitelist.txt)

### CODE REVIEW

**Comment by hwatersiv** [comment]
seems there was a failure to pass the checks. It appears it has to do with the package.json addtion:

```
Parsing in parallel...
Error: In package.json: Dependency @material-ui/core not in whitelist.
If you are depending on another `@types` package, do *not* add it to a `package.json`. Path mapping should make the import work.
If this is an external library that provides typings,  please make a pull request to types-publisher adding it to `dependenciesWhitelist.txt`.
    at checkPackageJsonDependencies (/home/vsts/work/1/s/node_modules/types-publisher/src/lib/definition-parser.ts:204:19)
    at combineDataForAllTypesVersions (/home/vsts/work/1/s/node_modules/types-publisher/src/lib/definition-parser.ts:99:37)
```

**Comment by petrbela** [comment]
@hwatersiv See the related [pull request](https://github.com/microsoft/types-publisher/pull/611)

File: types/mui-datatables/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "private": true,
+    "dependencies": {
+        "@material-ui/core": ">=3",
+        "csstype": "^2.0.0",
+        "indefinite-observable": "^1.0.1"
```
Comment (Favna) [inline types/mui-datatables/package.json]: Why are csstype and Indefinite-Observable a dependency now? I can sort of understand Material-ui/core due to your typings but I don't see these in your changes at all.

File: types/mui-datatables/index.d.ts [hunk 2]
```diff
@@ -3,10 +3,12 @@
 // Definitions by: Jeroen "Favna" Claassens <https://github.com/favna>
 //                 Ankith Konda <https://github.com/ankithkonda>
 //                 Herman "Von" Waters IV <https://github.com/hwatersiv>
+//                 Petr Bela <https://github.com/petrbela>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.1
+// TypeScript Version: 3.2
```
Comment (Favna) [inline types/mui-datatables/index.d.ts]: Please either describe why there is a requirement for TypeScript 3.2 or decrease it back down to 3.1. The only valid reason for the increase is if the typings use a structure that is part of the 3.2 spec but not the 3.1, and I sure don't see any of that.

**Review by favna — CHANGES_REQUESTED** [review]
Requesting some changes here

Edit: Also please either edit your original post unticking ` If you are making substantial changes, consider adding a tslint.json containing { "extends": "dtslint/dt.json" }.` or actually do what you claim to have done in adding a tslint.json containing the extension. 

Edit 2: Can you also add test code to `mui-datatables-tests.tsx` to cover these changes in test code?

File: types/mui-datatables/index.d.ts (hunk 2 above)
Comment (Favna) [reply types/mui-datatables/index.d.ts]: On further inspection I realize that you may have intended to increase the version number of the typings to reflect the version of the lib (currently on 2.2.0). That is line 1, not line 8. If this was your intention please make this change.

File: types/mui-datatables/package.json (hunk 1 above)
Comment (petrbela) [reply types/mui-datatables/package.json]: I might actually need some help here. The two dependencies should be removed. I only added them trying to troubleshoot the error I'm getting when I run `npm run tslint mui-datatables`:

```
Error: Errors in typescript@next for external dependencies:
../jss/css.d.ts(3,28): error TS2307: Cannot find module 'indefinite-observable'.
../jss/css.d.ts(4,26): error TS2307: Cannot find module 'csstype'.

    at /Users/petrbela/code/petrbela/DefinitelyTyped/node_modules/dtslint/bin/index.js:189:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/petrbela/code/petrbela/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
```

Same for `npm run test`. (Not sure why the CI didn't run here?)

`@material-ui/core` should pull in `csstype` and `indefinite-observable` as transitive dependencies via the `jss` package but it's not happening for some reason. Do you know why that might be?

File: types/mui-datatables/index.d.ts (hunk 2 above)
Comment (petrbela) [reply types/mui-datatables/index.d.ts]: Ah yes, sorry about that. I'll change the correct line.

**Comment by Favna** [comment]
@petrbela I'll take a loot at csstype and indefinite-observable tomorrow. For the time being can you commit that other change I requested about version number and try to cover the changes in the test file? To prevent the bot from closing the PR.

**Comment by petrbela** [comment]
Thanks, @Favna, I'll do!

**Comment by mviens** [comment]
I would really like to see these changes get added. What is the status of this @petrbela ?

**Pull request closed without merging**